tests/main/snapd-snap-transition-auto-install/task.yaml

```diff
@@ -0,0 +1,130 @@
+summary: Ensure auto-install based transition to snapd snap works.
+
+details: |
+    Ensure that we can transition classic systems to snapd snap
+    with auto-install of snapd snap from snapd deb or snapd in core
+    triggered by any snap install that is not of type os, base,
+    kernel, gadget or snapd. This change was introduced in PR14173.
+
+# Exclude Ubuntu Core systems - require proper remodeling
+# Simplify test by limiting system to ubuntu 22.04 amd64
+systems: [ubuntu-22.04-64]
+
+environment:
+    SNAPD_SRC/deb:  "deb"
+    SNAPD_SRC/core: "core"
+
+prepare: |
+    # remove all snaps to remove snapd
+    snap remove core
```

Comment:
I would check if it installed first, then I'll remove it

Reply:
added check